src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -806,16 +807,21 @@ public IEnumerable<SyntaxNodeOrToken> DescendantNodesAndTokensAndSelf(TextSpan s
         /// TODO: This should probably be reimplemented with <see cref="ChildThatContainsPosition"/>
         /// </remarks>
         /// <exception cref="ArgumentOutOfRangeException">This exception is thrown if <see cref="FullSpan"/> doesn't contain the given span.</exception>
+        [PerformanceSensitive("https://github.com/dotnet/roslyn/issues/26778", AllowCaptures = false)]
         public SyntaxNode FindNode(TextSpan span, bool findInsideTrivia = false, bool getInnermostNodeForTie = false)
         {
             if (!this.FullSpan.Contains(span))
             {
                 throw new ArgumentOutOfRangeException(nameof(span));
             }
 
-            var node = FindToken(span.Start, findInsideTrivia)
-                .Parent
-                .FirstAncestorOrSelf<SyntaxNode>(a => a.FullSpan.Contains(span));
+            SyntaxNode node;
+            using (PooledDelegates.GetPooledFunction((a, span) => a.FullSpan.Contains(span), span, out Func<SyntaxNode, bool> predicate))
```

Comment:
>using (PooledDelegates.GetPooledFunction((a, span) => a.FullSpan.Contains(span), span, out Func<SyntaxNode, bool> predicate)) [](start = 12, length = 125)

Could you please provide an explanation what are we trying to achieve here and how are we achieving that? Also, would it be possible to achieve the same or close effect by adding an overload of ```FirstAncestorOrSelf``` that takes a delegate with to parameters and the value to pass to that delegate as the second argument?

Reply:
This case is now replaced with an extension method that takes a TArg parameter.